wsmaster/codenvy-hosted-platform-api-impl/src/main/java/com/codenvy/api/workspace/LimitsCheckingWorkspaceManager.java

```diff
@@ -186,17 +190,47 @@ public WorkspaceImpl updateWorkspace(String id, Workspace update) throws Conflic
 
     /**
      * Checks that starting workspace won't exceed system RAM limit.
-     * Throws {@link BadRequestException} in the case of RAM constraint violation, otherwise
+     * Then, if previous check is passed, checks that starting workspace won't exceed user's started workspaces number limit.
+     * Throws {@link LimitExceededException} in the case of constraints violation, otherwise
      * performs {@code callback.call()} and returns its result.
      */
     @VisibleForTesting
-    <T extends WorkspaceImpl> T checkSystemRamAndPropagateStart(WorkspaceCallback<T> callback) throws ServerException,
-                                                                                                      NotFoundException,
-                                                                                                      ConflictException {
+    <T extends WorkspaceImpl> T checkLimitsAndPropagateStart(String nameSpace, WorkspaceCallback<T> callback)
+            throws ServerException, NotFoundException, ConflictException {
         if (systemRamInfoProvider.getSystemRamInfo().isSystemRamLimitExceeded()) {
             throw new LimitExceededException("Low RAM. Your workspace cannot be started until the system has more RAM available.");
         }
-        return callback.call();
+        return checkStartedWorkspacesNumberAndPropagateStart(nameSpace, callback);
+    }
+
+    /**
+     * Checks that starting workspace won't exceed user's started workspaces number limit.
+     * Throws {@link LimitExceededException} in the case of started workspaces number constraint violation, otherwise
+     * performs {@code callback.call()} and returns its result.
+     */
+    @VisibleForTesting
+    <T extends WorkspaceImpl> T checkStartedWorkspacesNumberAndPropagateStart(String nameSpace, WorkspaceCallback<T> callback)
+            throws ServerException, NotFoundException, ConflictException {
+        if (startedWorkspacesLimit < 0) {
+            return callback.call();
+        }
+
+        // It is important to lock in this place because:
+        // if started workspaces number limit is 10 and user has 9 started workspaces, then if he sends 2 separate requests
+        // to start a workspace, it may start both of them, because started workspaces number check is not atomic one.
+        final Lock lock = CREATE_LOCKS.get(nameSpace);
```

Comment:
usage in `checkCountAndPropagateCreation()` is clear. But you should not use CREATE_LOCK here. You should create new one and use it. Or rename existing one to something common. I prefer creating new one lock because it won't lock workspaces creating on workspaces starting and vice versa